src/types/EIP712Types.sol

```diff
@@ -0,0 +1,215 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.27;
+
+// Message signed by the sponsor that specifies the conditions under which their
+// tokens can be claimed; the specified arbiter verifies that those conditions
+// have been met and specifies a set of beneficiaries that will receive up to the
+// specified amount of tokens.
+struct Compact {
+    address arbiter; // The account tasked with verifying and submitting the claim.
+    address sponsor; // The account to source the tokens from.
```

Comment:
a few reasons:
- the allocator also signs the payload, so that way one payload can't be replayed for multiple sponsors
- by including this in the payload, the resultant claim hash can uniquely identify the compact in question
- you can run validation on the payload without the final signature (say a filler wants to provide a quote and wants to ensure the balance is available, but the allocator doesn't want to leak the full signed payload prematurely)